src/components/StatusUpdateForm/useStatusUpdateForm.ts

```diff
@@ -0,0 +1,545 @@
+      setImages((prevState) => {
+        if (!prevState.data[id]) return prevState;
+        prevState.data[id].url = url;
+        prevState.data[id].state = 'finished';
+        return { ...prevState };
+      });
+    } catch (e) {
+      console.warn(e);
+
+      setImages((prevState) => {
+        if (!prevState.data[id]) return prevState;
+        logErr(e, 'upload-image');
+        prevState.data[id].state = 'failed';
+        return { ...prevState };
+      });
+    }
+  }, []);
+
+  const uploadFile = useCallback(async (id: string, file: FileUploadState) => {
+    setFiles((prevState) => {
+      if (!prevState.data[id]) return prevState;
+      prevState.data[id].state = 'uploading';
+      return { ...prevState, data: { ...prevState.data } };
+    });
+
+    try {
+      const { file: url } = await client.files.upload(file.file as File);
+      setFiles((prevState) => {
+        if (!prevState.data[id]) return prevState;
+        prevState.data[id].url = url;
+        prevState.data[id].state = 'finished';
+        return { ...prevState, data: { ...prevState.data } };
+      });
+    } catch (e) {
+      console.warn(e);
+      setFiles((prevState) => {
+        if (!prevState.data[id]) return prevState;
+        logErr(e, 'upload-file');
+        prevState.data[id].state = 'failed';
+        return { ...prevState, data: { ...prevState.data } };
+      });
+    }
+  }, []);
+
+  const uploadNewFiles = useCallback((files: Blob[] | File[] | FileList) => {
+    for (let i = 0; i < files.length; i += 1) {
+      const file = files[i];
+      if (file.type.startsWith('image/')) {
+        uploadNewImage(file);
+      } else if (file instanceof File) {
+        uploadNewFile(file);
+      }
+    }
+  }, []);
+
+  const removeImage = useCallback((id: string) => {
+    setImages((prevState) => {
+      prevState.order = prevState.order.filter((oid) => id !== oid);
+      delete prevState.data[id];
+      return { ...prevState };
+    });
+  }, []);
+
+  const removeFile = useCallback((id: string) => {
+    // eslint-disable-next-line sonarjs/no-identical-functions
+    setFiles((prevState) => {
+      prevState.order = prevState.order.filter((oid) => id !== oid);
+      delete prevState.data[id];
+      return { ...prevState };
+    });
+  }, []);
+
+  useEffect(() => {
+    images.order
+      .filter((id) => !reqInProgress.current[id] && images.data[id].state === 'uploading')
+      .forEach(async (id) => {
+        reqInProgress.current[id] = true;
+        await uploadImage(id, images.data[id]);
+        delete reqInProgress.current[id];
+      });
+  }, [images.order]);
+
+  useEffect(() => {
+    files.order
+      .filter((id) => !reqInProgress.current[id] && files.data[id].state === 'uploading')
+      .forEach(async (id) => {
+        reqInProgress.current[id] = true;
+        await uploadFile(id, files.data[id]);
+        delete reqInProgress.current[id];
+      });
+  }, [files.order]);
```

Comment:
These two look identical, might be a good idea to create one parametrised function for it.

Reply:
at this point refactoring these two parts actually makes the code harder to follow IMO

Reply:
btw, please don't force push to a branch which is not local anymore